### PR TITLE
Clarify documentation load functions

### LINKED ISSUE #2349: plot EDS hypermap from .bcf
I have loaded the bruker hypermap data in the notebook. Somehow I could not plot it. I just verify the plot function works fine for .emi file from FEI. My microscope is Hitachi HD2700 STEM. I am wondering whether we could not plot it in the bcf file. Could you please help me to fix it out? @sem-geologist 

Here is my code:
```python
%matplotlib qt
import hyperspy.api as hs
import numpy as np
import matplotlib.pyplot as plt
s = hs.load("1.bcf")
s.plot()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-886abc61b753> in <module>
----> 1 s.plot()

AttributeError: 'list' object has no attribute 'plot'
```

### PR DESCRIPTION
**Pull request by ericpre**
Recently, a few users have asked about the following error message (for example https://github.com/hyperspy/hyperspy-demos/issues/55#issuecomment-607525445 and #2349):

```python
TypeError Traceback (most recent call last)
in
9 # <Signal2D, title: , dimensions: (|512, 512)>,
10 # <Signal2D, title: , dimensions: (|512, 512)>]
---> 11 s[0].plot()
```

### Progress of the PR
- [x] Clarify output of the load functions in its docstring and the user guide,
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] move the "interactive operations and ROI" section to its own page to give it more visibility,
- [x] fix a couple of warnings when building the documentation,
- [x] ready for review.

### CODE REVIEW

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+Interactive Operations and Region of Interest (ROI)
+***************************************************
+
+.. _interactive-label:
+
+Interactive operations
+----------------------
+
+
+The function :py:func:`~.interactive.interactive` ease the task of defining
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: --> simplifies the definition of....

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+operations that are automatically updated when an event is triggered. By
+default it recomputes the operation when data or the axes of the original
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: By default the operation is recomputed when the data or...

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+signal changes.
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: signal is changed.

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    >>> s = hs.signals.Signal1D(np.arange(10.))
+    >>> ssum = hs.interactive(s.sum, axis=0)
+    >>> ssum.data
+    array([45.0])
+    >>> s.data /= 10
+    >>> s.events.data_changed.trigger(s)
+    >>> ssum.data
+    array([ 4.5])
+
+The interactive operations can be chained.
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: Interactive operations can be performed in a chain.

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 5]
```diff
@@ -0,0 +1,257 @@
+
+
+Interactive Operations and Region of Interest (ROI)
+***************************************************
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst:4]: Interactive and Region of Interest (ROI) Operations

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    >>> s.data *= 10
+    >>> s.events.data_changed.trigger(obj=s)
+    >>> ssum_mean.data
+    array([ 300.,  330.,  360.,  390.])
+
+.. _roi-label:
+
+Region Of Interest (ROI)
+------------------------
+
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst:54]: ROIs can be defined to select part of any compatible signal and may be applied either to the navigation or to the signal axes.

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+A number of different ROIs are available:
+
+* :py:class:`~.roi.Point1DROI`
+* :py:class:`~.roi.Point2DROI`
+* :py:class:`~.roi.SpanROI`
+* :py:class:`~.roi.RectangularROI`
+* :py:class:`~.roi.CircleROI`
+* :py:class:`~.roi.Line2DROI`
+
+Once created, a ROI can be used to return a part of any compatible signal:
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: Once created, an ROI can be applied to the signal:

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+    >>> roi = hs.roi.RectangularROI(left=3, right=7, top=2, bottom=5)
+    >>> sr = roi(s)
+    >>> sr
+    <Signal1D, title: , dimensions: (4, 3|10)>
+    >>> imr = roi(im)
+    >>> imr
+    <Signal2D, title: , dimensions: (|4, 3)>
+
+ROIs can also be used :ref:`interactively <interactive-label>` with widgets.
+The following examples shows how to interactively apply ROIs to an image. Note
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: The following example shows....

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+.. figure::  images/roi_line2d_holo.png
+  :align:   center
+  :width:   500
+
+.. code-block:: python
+
+    >>> roi.angle(axis='y')
+    -100.97166759025453
+
+By default output of the method is in degrees, though radians can be selected as follows:
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: The default output of the method...

File: doc/user_guide/interactive_operations_ROIs.rst [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+
+.. code-block:: python
+
+    >>> roi.angle(axis='vertical', units='radians')
+    -1.7622880506791903
+
+Conveniently, :meth:`~.roi.Line2DROI.angle` can be used to rotate image to align
```
Comment (dnjohnstone) [inline doc/user_guide/interactive_operations_ROIs.rst]: can be used to rotate an image...

File: doc/user_guide/io.rst [hunk 11]
```diff
@@ -19,9 +20,12 @@ image ascent.jpg you can type:
 
     >>> s = hs.load("ascent.jpg")
 
-If the loading was successful, the variable ``s`` contains a generic
-:py:class:`~.signal.BaseSignal`, a :py:class:`~._signals.signal1d.Signal1D` or
-an :py:class:`~._signals.signal2d.Signal2D`.
+If the loading was successful, the variable ``s`` contains a HyperSpy signal
```
Comment (dnjohnstone) [inline doc/user_guide/io.rst]: If loading...

File: doc/user_guide/io.rst [hunk 12]
```diff
@@ -19,9 +20,12 @@ image ascent.jpg you can type:
...
+or a signal of the :ref:`HyperSpy extensions <hyperspy_extensions-label>`
+- see available :ref:`signal subclasses <transforming_signal-label>` for more
+information.
+If the loaded file contains some severals dataset, The ``load`` functions will
```
Comment (dnjohnstone) [inline doc/user_guide/io.rst]: ...contains several datasets, The load function will.....

File: doc/user_guide/io.rst [hunk 13]
```diff
@@ -812,7 +816,7 @@ the data size in memory.
 .. note::
 
     Pruned Velox EMD files only contain the spectrum image in a proprietary
-    format that HyperSpy cannot read. Therefore, don't prune FEI EMD files in 
+    format that HyperSpy cannot read. Therefore, don't prune FEI EMD files in
```
Comment (dnjohnstone) [inline doc/user_guide/io.rst]: ....files if....

File: hyperspy/io.py [hunk 14]
```diff
@@ -60,6 +60,9 @@ def load(filenames=None,
     Bruker bcf and spx, FEI ser and emi, SEMPER unf, EMD, EDAX spd/spc,
     tif, and a number of image formats.
 
+    Depending ont the number of dataset in the file to load, this function will
```
Comment (dnjohnstone) [inline hyperspy/io.py]: ...number of datasets to load in the file, this...

**Review by dnjohnstone — CHANGES_REQUESTED** [review]
Just some minor tweaks - then it's a please merge from me.

**Comment by jlaehne** [comment]
LGTM

**Comment by francisco-dlp** [comment]
LGTM too. Could you solve the conflicts?

**Comment by ericpre** [comment]
Done.

**Pull request merged**